jdaviz/configs/imviz/plugins/orientation/orientation.py

```diff
@@ -767,52 +771,82 @@ def link_image_data(app, align_by='pixels', wcs_fallback_scheme=None, wcs_fast_a
     ndim_range = range(2)  # We only support 2D
 
     for i, data in enumerate(app.data_collection):
-        # 1. Do not link with self.
-        # 2. We are not touching any existing Subsets or Table. They keep their own links.
-        # 3. Links already exist for this entry and we're not changing the type.
-        # 4. We are not touching fake WCS layers in pixel linking.
-        # 5. We are not touching data without WCS in WCS linking.
-        if ((i == iref) or (not layer_is_2d(data)) or (data in data_already_linked) or
-                (align_by == "pixels" and data.meta.get(_wcs_only_label)) or
-                (align_by == "wcs" and not hasattr(data.coords, 'pixel_to_world'))):
+
+        # Do not link with self or existing links.
+        if i == iref or data in data_already_linked:
             continue
 
-        ids1 = data.pixel_component_ids
-        new_links = []
-        try:
-            if align_by == 'pixels':
-                new_links = [LinkSame(ids0[i], ids1[i]) for i in ndim_range]
-            else:  # wcs
-                wcslink = WCSLink(data1=refdata, data2=data, cids1=ids0, cids2=ids1)
-                if wcs_fast_approximation:
-                    try:
-                        new_links = [wcslink.as_affine_link()]
-                    except NoAffineApproximation:  # pragma: no cover
-                        new_links = [wcslink]
-                else:
-                    new_links = [wcslink]
-        except Exception as e:  # pragma: no cover
-            if align_by == 'wcs' and wcs_fallback_scheme == 'pixels':
-                try:
+        # Special handling of Catalogs: only link RA/Dec to reference WCS,
+        # and use ComponentLink instead of WCSLink because Catalogs do not
+        # have coords (wcs) attribute.
+        if data.meta.get('_importer') == 'CatalogImporter':
+            comp_labels = [data.components[i].label for i in range(len(data.components))]

```

Comment:
```suggestion
            comp_labels = data.component_ids()
```
I think this accomplishes the same thing.